test/include/dlaf_test/matrix/util_matrix_blas.h

```diff
@@ -0,0 +1,60 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#include "blas.hh"
+#include "dlaf/matrix.h"
+#include "dlaf_test/matrix/util_matrix.h"
+#include "dlaf_test/util_types.h"
+
+namespace dlaf {
+namespace matrix {
+namespace test {
+using namespace dlaf_test;
+
+/// @brief Sets the elements of the matrix.
+///
+/// The (i, j)-element of the matrix is set to el({i, j}) if op == NoTrans,
+///                                          el({j, i}) if op == Trans,
+///                                          conj(el({j, i})) if op == ConjTrans.
+/// @pre el argument is an index of type const GlobalElementIndex& or GlobalElementIndex.
+/// @pre el return type should be T.
+template <template <class, Device> class MatrixType, class T, class ElementGetter>
+void set(MatrixType<T, Device::CPU>& mat, ElementGetter el, blas::Op op) {
+  switch (op) {
+    case blas::Op::NoTrans:
+      set(mat, el);
+      break;
+
+    case blas::Op::Trans: {
+      auto op_el = [el](GlobalElementIndex i) {
```

Comment:
You are passing `el` to the lambda by copy. Since there is no concurrency problem, I would pass it by reference, mainly because in this way you allow the `ElementGetter` to have its internal status and keep it for its lifetime over multiple calls (even if it is not a big deal).

In case you agree, this is not the only occurrence (see also `util_tile_blas.h`).